File: oj_server/questions/1/java/tail.java

// ======== TAIL ========
@FunctionalInterface
interface TestFunc { boolean run(); }

static class TestCase {
    TestFunc f; String in;
    TestCase(TestFunc f, String in) { this.f = f; this.in = in; }
}

static boolean assertEqual(boolean a, boolean e) { return a == e; }

static int saveResult(List<TestCase> tests, String path) {
    int total = tests.size(), passed = 0;
    String firstFail = "";
    for (TestCase t : tests) {
        if (t.f.run()) passed++;
        else if (firstFail.isEmpty()) firstFail = t.in;
    }
    try (FileWriter w = new FileWriter(path)) {
        w.write(passed + " " + total + " " + firstFail);
    } catch (IOException _){ return 1; }
    return passed == total ? 0 : 1;
}

public static void main(String[] args) {
    if (args.length < 1) System.exit(1);
    String path = args[0];
    List<TestCase> tests = Arrays.asList(
        new TestCase(() -> assertEqual(new Solution().isPalindrome(0), true), "0"),
        new TestCase(() -> assertEqual(new Solution().isPalindrome(-1), false), "-1"),
        new TestCase(() -> assertEqual(new Solution().isPalindrome(5), true), "5"),
        new TestCase(() -> assertEqual(new Solution().isPalindrome(10), false), "10"),
        new TestCase(() -> assertEqual(new Solution().isPalindrome(121), true), "121"),
        new TestCase(() -> assertEqual(new Solution().isPalindrome(-121), false), "-121"),
        new TestCase(() -> assertEqual(new Solution().isPalindrome(12321), true), "12321"),
        new TestCase(() -> assertEqual(new Solution().isPalindrome(1231), false), "1231"),
        new TestCase(() -> assertEqual(new Solution().isPalindrome(1001), true), "1001"),
        new TestCase(() -> assertEqual(new Solution().isPalindrome(2147447412), true), "2147447412")
    );
    System.exit(saveResult(tests, path));
}

File: oj_server/questions/1/go/tail.go

func assertEqual(actual, expected bool) bool { return actual == expected }

type TestCase struct {
    testFunc func() bool
    inputStr string
}

func saveResult(tests []TestCase, resultPath string) int {
    total, passed := len(tests), 0
    firstFail := ""
    for _, t := range tests {
        if t.testFunc() {
            passed++
        } else if firstFail == "" {
            firstFail = t.inputStr
        }
    }
    f, err := os.Create(resultPath)
    if err != nil { return 1 }
    defer f.Close()
    fmt.Fprintf(f, "%d %d %s", passed, total, firstFail)
    if passed == total { return 0 }
    return 1
}

func main() {
    if len(os.Args) < 2 { os.Exit(1) }
    path := os.Args[1]
    tests := []TestCase{
        {func() bool { return assertEqual(Solution{}.isPalindrome(0), true) }, "0"},
        {func() bool { return assertEqual(Solution{}.isPalindrome(-1), false) }, "-1"},
        {func() bool { return assertEqual(Solution{}.isPalindrome(5), true) }, "5"},
        {func() bool { return assertEqual(Solution{}.isPalindrome(10), false) }, "10"},
        {func() bool { return assertEqual(Solution{}.isPalindrome(121), true) }, "121"},
        {func() bool { return assertEqual(Solution{}.isPalindrome(-121), false) }, "-121"},
        {func() bool { return assertEqual(Solution{}.isPalindrome(12321), true) }, "12321"},
        {func() bool { return assertEqual(Solution{}.isPalindrome(1231), false) }, "1231"},
        {func() bool { return assertEqual(Solution{}.isPalindrome(1001), true) }, "1001"},
        {func() bool { return assertEqual(Solution{}.isPalindrome(2147447412), true) }, "2147447412"},
    }
    os.Exit(saveResult(tests, path))
}